pkg/kubecost/allocation.go

```diff
@@ -107,9 +107,52 @@ type RawAllocationOnlyData struct {
 	RAMBytesUsageMax float64 `json:"ramByteUsageMax"`
 }
 
-// PVUsage contains the byte hour usage
+// PV is a map of Disk Asset Identifiers to the
+// usage of them by an Allocation as recorded in a PVAllocation
+type PV map[PVKey]*PVAllocation
+
+// Clone creates a deep copy of a PV
+func (pv *PV) Clone() PV{
+	if pv == nil || *pv == nil {
+		return nil
+	}
+	apv := *pv
+	clonePV := PV{}
+	for k, v := range apv{
+		clonePV[k] = v
+	}
+	return clonePV
+}
+
+// Add adds contents of that to the calling PV
+func (pv *PV) Add(that PV) PV{
+	apv := pv.Clone()
+	if that != nil {
+		if apv == nil {
+			apv = PV{}
+		}
+		for pvKey, thatPVAlloc := range that {
+			apvAlloc, ok := apv[pvKey]
+			if !ok {
+				apvAlloc = &PVAllocation{}
+			}
+			apvAlloc.Cost += thatPVAlloc.Cost
+			apvAlloc.ByteHours += thatPVAlloc.ByteHours
+			apv[pvKey] = apvAlloc
+		}
+	}
+	return apv
+}
+
+// PVKey for identifying Disk type assets
+type PVKey struct {
+	Cluster string `json:"cluster"`
+	Name    string `json:"name"`
+}
+
+// PVAllocation contains the byte hour usage
 // and cost of an Allocation for a single PV
-type PVUsage struct {
+type PVAllocation struct {
```

Comment:
Should we consider adding "Name" here? I know it's in the map, but do you see us wanting to use this struct standalone at some point?

Reply:
I don't really see it as ever becoming a stand alone, at this point I think it would be better to keep it more light weight, if we do need to add it in the future its not like it will be particularly difficult.